File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        tracker = {}
        
        for index, num in enumerate(nums):
            diff = target - num
            if diff in tracker:
                return [tracker[diff], index]
            else:
                tracker[num] = index
                
        return [-1,-1]
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap <Integer,Integer> tracker = new HashMap<>();
        

        
        for(int i=0; i < nums.length; i++){
            
            int diff = target - nums[i];
            
            if(tracker.containsKey(diff)){
                return new int[] {tracker.get(diff), i};
            }
            else{
                tracker.put(nums[i],i);
            }
            
        }
        
        return new int[]{-1,-1};
    }
}